test/extended/util/framework.go

```diff
@@ -219,7 +221,20 @@ func WaitForADeployment(client kclient.ReplicationControllerInterface,
 		defer w.Stop()
 
 		for {
-			val, ok := <-w.ResultChan()
+			timeout := make(chan bool, 1)
+			go func() {
+				time.Sleep(10 * time.Second)
```

Comment:
The problem was that `<-w.ResultChan() was never returning.  The extended test framework eventually killed the test after an hour.  There was some discussion in my PR as to why that was the case.